### PR TITLE
docs: add a table with an overview how propts,resources,tools,sampling is generally used

### PR DESCRIPTION
**Pull request by dsp-ant**
image

### CODE REVIEW

File: docs/spec/_index.md [hunk 1]
```diff
@@ -32,6 +32,17 @@ On top of the base protocol, MCP introduces these unique primitives:
...
 * **Sampling**: *servers* can ask the *client* to sample from the LLM, which allows servers to implement agentic behaviors without having to implement sampling themselves. This also allows the client to combine the sampling request with *all of the other context it has*, making it much more intelligent—while avoiding needlessly exfiltrating information to servers.
 
+Each primitive can be summarized in the following control hierarchy:
+
+| Primitive | Control                | Description                                        | Example                      |
+|-----------|------------------------|----------------------------------------------------|------------------------------|
+| Prompts   | User-controlled        | Interactive templates invoked by user choice       | Slash commands, menu options |
+| Resources | Application-controlled | Contextual data attached and managed by the client | File contents, git history   |
+| Tools     | Model-controlled       | Functions exposed to the LLM to invoke as needed   | API calls, file operations   |
+| Sampling  | Server-initiated       | LLM inference requests from server to client       | Generating completions       |
```
Comment (jspahrsummers) [inline docs/spec/_index.md]: Suggested better example:
```suggestion
| Sampling  | Server-initiated       | LLM completion requests from server to client       | Generating AI summaries      |
```

File: docs/spec/_index.md [hunk 2]
```diff
@@ -32,6 +32,17 @@ On top of the base protocol, MCP introduces these unique primitives:
...
 * **Sampling**: *servers* can ask the *client* to sample from the LLM, which allows servers to implement agentic behaviors without having to implement sampling themselves. This also allows the client to combine the sampling request with *all of the other context it has*, making it much more intelligent—while avoiding needlessly exfiltrating information to servers.
 
+Each primitive can be summarized in the following control hierarchy:
+
+| Primitive | Control                | Description                                        | Example                      |
+|-----------|------------------------|----------------------------------------------------|------------------------------|
+| Prompts   | User-controlled        | Interactive templates invoked by user choice       | Slash commands, menu options |
+| Resources | Application-controlled | Contextual data attached and managed by the client | File contents, git history   |
+| Tools     | Model-controlled       | Functions exposed to the LLM to invoke as needed   | API calls, file operations   |
```
Comment (jspahrsummers) [inline docs/spec/_index.md]: I suggest emphasizing the "write" side of tools:

```suggestion
| Tools     | Model-controlled       | Functions exposed to the LLM to take actions    | API POST requests, file writing   |
```

Maybe we can also allude to the idea of clients supporting tools that can just run `resource/list` and `resource/read` directly?

**Comment by jspahrsummers** [comment]
Do we want to include a flowchart similar to the following (with appropriate cleanup/simplification) somewhere too?

```mermaid
flowchart LR
    B{{What should the server do?}}
    
    B -->|Provide context to LLM| C[Define resources in server]
    C --> C1[Static data]
    C --> C2[Dynamic data]
    C --> C3[Real-time updates]
    
    B -->|Enable agentic behavior| D[Define tools in server]
    D --> D1[Actions for LLM to invoke]
    D --> D2[Integrate with external systems]
    
    B -->|Define reusable LLM interactions| E[Define prompts in server]
    E --> E1[Templated prompts]
    E --> E2[Predefined LLM tasks]
    
    B -->|Custom LLM processing| F[Server sampling from client]
    F --> F1[Recursive LLM operations]
    F --> F2[Data preprocessing]
```

**Comment by dsp-ant** [comment]
> Do we want to include a flowchart similar to the following (with appropriate cleanup/simplification) somewhere too?
> 
> ```mermaid
> flowchart LR
>     B{{What should the server do?}}
>     
>     B -->|Provide context to LLM| C[Define resources in server]
>     C --> C1[Static data]
>     C --> C2[Dynamic data]
>     C --> C3[Real-time updates]
>     
>     B -->|Enable agentic behavior| D[Define tools in server]
>     D --> D1[Actions for LLM to invoke]
>     D --> D2[Integrate with external systems]
>     
>     B -->|Define reusable LLM interactions| E[Define prompts in server]
>     E --> E1[Templated prompts]
>     E --> E2[Predefined LLM tasks]
>     
>     B -->|Custom LLM processing| F[Server sampling from client]
>     F --> F1[Recursive LLM operations]
>     F --> F2[Data preprocessing]
> ```

I am a bit worried that unless you are super into the topic, this is more confusing than clarifying. We could add an "in-depth appendix or so", but for now I'd keep it out of the docs unless you think it would be super useful.

**Pull request merged**